.github/workflows/ci-preview.yml

```diff
@@ -0,0 +1,29 @@
+name: Generate Preview (CI)
+
+concurrency:
+  group: ci-${{ github.event.number }}
```

Comment:
I think just `group: ci` is fine here, the event number concurrency group is useful when multiple jobs could stem from logically related events like pushes to a PR. But here this should just be happening once on each merge to `main`.